src/main/java/io/lettuce/core/protocol/EventListenerCommand.java

```diff
@@ -0,0 +1,88 @@
+package io.lettuce.core.protocol;
+
+import io.lettuce.core.RedisCommandExecutionException;
+import io.lettuce.core.RedisCommandListener;
+import io.lettuce.core.models.events.CommandFailedEvent;
+import io.lettuce.core.models.events.CommandStartedEvent;
+import io.lettuce.core.models.events.CommandSucceededEvent;
+import io.lettuce.core.output.CommandOutput;
+import io.netty.buffer.ByteBuf;
+
+/**
+ * @author sokomishalov
+ */
+public class EventListenerCommand<K, V, T> implements RedisCommand<K, V, T>, DecoratedCommand<K, V, T> {
+
+    private final CommandStartedEvent<K, V, T> startedEvent;
+    private final RedisCommandListener listener;
+
+    public EventListenerCommand(CommandStartedEvent<K, V, T> startedEvent, RedisCommandListener listener) {
```

Comment:
Let's keep only a reference to the command and the context.

We should be able to make this class package-private.

Reply:
Fixed.
I think it could be even with private inner visibility inside `CommandListenerWriter`